score_models/solver.py

```diff
@@ -0,0 +1,154 @@
+from abc import ABC, abstractmethod
+
+import torch
+import numpy as np
+from tqdm import tqdm
+
+
+class Solver(ABC):
+    """
+    Base solver for a stochastic differential equation (SDE).
+
+    The SDE defines the differential element ``dx`` for the stochastic process
+    ``x``. Both the forward and reverse processes are defined using the
+    coefficients and score provided. This class represents essentially any SDE
+    of the form: :math:`dx = f(t, x) dt + g(t, x) dw`.
+    """
+
+    def __init__(self, score, corrector=False, **kwargs):
+        self.score = score
+        self.corrector = corrector
+        self.corrector_epsilon = kwargs.get("corrector_epsilon", lambda t: 0.1)
```

Comment:
This is not how the corrector step is computed. A user needs to define an SNR (a simple float), and the corrector step is computed as epsilon = (SNR * sigma(t))**2, where sigma(t) comes from the SDE.

Reply:
Ah ok, I wasn't sure if that was universal, or just a common convention so I left it as an arbitrary function. I have now updated it to the formula you provided.